pkg/controller/backup/backup_control.go

```diff
@@ -53,12 +53,14 @@ type defaultBackupControl struct {
 // UpdateBackup executes the core logic loop for a Backup.
 func (bc *defaultBackupControl) UpdateBackup(backup *v1alpha1.Backup) error {
 	backup.SetGroupVersionKind(controller.BackupControllerKind)
-	if err := bc.addProtectionFinalizer(backup); err != nil {
-		return err
-	}
+	if backup.Spec.CleanData {
```

Comment:
this check seems not necessary and IMO we should always try to add/remove finalizer, for example, users may change this field from `true` to `false`